packages/node/src/e2e/e2e-test.spec.ts

```diff
@@ -36,157 +36,201 @@ config({
   silent: process.env.CI === "true",
 });
 
-// This first test just saves the trouble of looking for a library failure when
-// the environment wasn't properly set.
-describe("Environment", () => {
-  it("contains the expected environment variables", () => {
-    expect(process.env.E2E_TEST_CLIENT_ID).not.toBeUndefined();
-    expect(process.env.E2E_TEST_CLIENT_SECRET).not.toBeUndefined();
-    expect(process.env.E2E_TEST_IDP_URL).not.toBeUndefined();
-  });
-});
-
-describe("Authenticated fetch", () => {
-  it("properly sets up session information", async () => {
-    const session = new Session();
-    await session.login({
-      clientId: process.env.E2E_TEST_CLIENT_ID,
-      clientSecret: process.env.E2E_TEST_CLIENT_SECRET,
-      oidcIssuer: process.env.E2E_TEST_IDP_URL,
-    });
-    expect(session.info.isLoggedIn).toEqual(true);
-    expect(session.info.sessionId).not.toBeUndefined();
-    expect(session.info.webId).not.toBeUndefined();
-  });
-
-  it("can fetch a public resource when logged in", async () => {
-    const session = new Session();
-    await session.login({
-      clientId: process.env.E2E_TEST_CLIENT_ID,
-      clientSecret: process.env.E2E_TEST_CLIENT_SECRET,
-      oidcIssuer: process.env.E2E_TEST_IDP_URL,
-    });
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    const publicResourceUrl = session.info.webId!;
-    const response = await session.fetch(publicResourceUrl);
-    expect(response.status).toEqual(200);
-    await expect(response.text()).resolves.toContain(
-      ":PersonalProfileDocument"
-    );
-  });
-
-  it("can fetch a private resource when logged in", async () => {
-    const session = new Session();
-    await session.login({
-      clientId: process.env.E2E_TEST_CLIENT_ID,
-      clientSecret: process.env.E2E_TEST_CLIENT_SECRET,
-      oidcIssuer: process.env.E2E_TEST_IDP_URL,
+type OidcIssuer = string;
+type ClientId = string;
+type ClientSecret = string;
+type Pod = string;
+type AuthDetails = [Pod, OidcIssuer, ClientId, ClientSecret];
+// Instructions for obtaining these credentials can be found here:
+// https://github.com/inrupt/solid-client-authn-js/blob/1a97ef79057941d8ac4dc328fff18333eaaeb5d1/packages/node/example/bootstrappedApp/README.md
+const serversUnderTest: AuthDetails[] = [
```

Comment:
Could we maybe use the same pattern as in @inrupt/solid-client-js here?

That is: using an array of objects instead of an array of arrays:
https://github.com/inrupt/solid-client-js/blob/09024b5a08f8ab00bd503a36e67cc1efb50e684b/src/e2e-browser/e2e.testcafe.ts#L76-L80

I think it's a bit cleaner than positional arguments, but I don't mind too much. Just a suggestion.

Reply:
unfortunately that's not possible, because these are two different things: in this case, we have to conform to Jest's `describe.each` signature, and an array of array is what is expected. Note that the file you're pointing to contains browser tests, which run with TestCafé, and not Jest, which is why we use a different syntax.

Reply:
Oh, sad. Well, c'est la vie!